### PR TITLE
feat(Patient financial activities) add movement value and total on the pan

### LINKED ISSUE #4302: Add the value of stock movements given to a patient to the patient's financial activity page
This issue can be tackled at the same time as #4300.

After #4269 landed, we now have the ability to see the stock movements to the patient.  It would be helpful if we could also have the value ($/FC) of the stock movements visible on each row, and the total value visible on the panel. 

This would provide a quick view of :
  1. What the patient was billed.
  2. What the patient paid.
  3. What the patient received.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #4302

stock_value

### CODE REVIEW

File: client/src/js/components/bhPatientFinancialActivity.js [hunk 1]
```diff
@@ -67,6 +68,7 @@ function PatientFinancialActivityCtrl(Patients, moment, Session, Constants, $q)
         stockData.forEach((item, index) => {
           if (index < limitRecent) {
             dataMovement.push(item);
+            $ctrl.dataMovementTotalValue += item.value;
```
Comment (jniles) [inline client/src/js/components/bhPatientFinancialActivity.js]: This should sum all movements.  We need to know the total value, not just the value of 5 displayed.

File: server/controllers/medical/patients/index.js [hunk 2]
```diff
@@ -699,10 +699,17 @@ function getStockMovements(req, res, next) {
 
 function stockMovementByPatient(patientUuid) {
   const sql = `
-    SELECT DISTINCT(sm.document_uuid) AS document_uuid, sm.depot_uuid, sm.date, map.text AS reference_text
+      SELECT DISTINCT BUID(sm.document_uuid) AS document_uuid, 
+      BUID(sm.depot_uuid) as depot_uuid,
+      sm.unit_cost,
+      (sm.quantity * sm.unit_cost) as value,
```
Comment (jniles) [inline server/controllers/medical/patients/index.js:705]: :+1:

File: client/src/modules/patients/record/bh-patient-financial-activity.html [hunk 3]
```diff
@@ -99,22 +99,31 @@ <h5 style="margin-top:0;">
...
               <tr ng-repeat="dataMovement in $ctrl.dataMovement">
                 <td>{{ $index + 1 }}</td>
                 <td><strong> {{ dataMovement.reference_text }} </strong></th>
+                <td>{{dataMovement.inventory_name}}</td>
                 <td> {{ dataMovement.date | date : "dd/MM/yyyy HH:mm:ss" }} </td>
+                <td><strong> {{ dataMovement.value | currency:$ctrl.enterpriseCurrencyId }} </strong></th>
+                
+              </tr>
+              <tr  ng-if="!$ctrl.noStockMovement">
+                <td colspan="4" class="text-right" translate>FORM.LABELS.TOTAL</td>
```
Comment (jniles) [inline client/src/modules/patients/record/bh-patient-financial-activity.html]: Should this be colspan=5?

**Review by jniles — COMMENTED** [review]
This is close!  We need to sum all stock movement though, not just the ones displayed.

**Comment by jniles** [comment]
@jeremielodi I see you re-requested a review, but it still looks like you are only summing the values of the displayed stock lots.  We need to know the value of all lots given to a patient.

**Comment by jeremielodi** [comment]
I was just afraid to display the total of all movement as long as the user is just seeing just few rows

**Review by jniles — APPROVED** [review]
LGTM.

bors r+

**Pull request merged**